### ISSUE TITLE
List dynamic height

### ISSUE THREAD

**Issue by JakubNajvarek** [appsmithorg/appsmith#31616]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

It would be very nice if there would be a possibility to have "Auto height" or dynamic height option on a list component (So that the height would depend on number of items).

### Why should this be worked on?

I think it would really help so that the list looks better for the app users and i also think that its a feature thats used on other components so other so it wouldnt be that hard to implement it (I hope).

**Comment by Nikhil-Nandagopal** [comment]
@JakubNajvarek thank you for the feature request!

**Comment by okletsov** [comment]
Yes, please add the auto-height feature if you have cycles. I have a table which can have varying number of records returned from DB based on the time of day and the day of week. It is ranging anywhere from 0 to 20+ items. The list needs to live on the page with other components. As you can probably guess, having fixed width wouldn't work in this case.

**Comment by matangilad-fiverr** [comment]
Hi guys, any update on that? I have a similar use case where I need to have a list which is dynamic in size, not only by data returned from the data source but also can be changed by the user adding/removing elements from it.

**Comment by trmc10** [comment]
Hello all, having the ability for the List component to be auto-height, based on its content, is a critical need for our application. Similar to others, this list will expand and contract depending on data and user interaction, and the current fixed height/pagination paradigm does not allow for the required interactions we need.

**Comment by chawlz7** [comment]
We could really use this ability ASAP. We need to use two lists on an interface and not being able to dynamically size them based on the number of items on the list makes it incredibly wonky/unusable. This feels like an easy fix, I'm not sure why it doesn't already work this way.

**Comment by umbearto** [comment]
List component is imho the most flexible and allows creating really cool looking lists.
Having dynamic height on every "card" would allow to create awesome collapsible lists :)

**Comment by PatrickCoyle9** [comment]
Is there any update to this?

**Comment by automatron3k** [comment]
I want to second this. Collapsable cards are needed, specially considering that the list widget accepts all other widgets inside, which converts it automatically in the go-to widget for lots of use-cases, but not so much if the height is fixed, limiting radically the dynamic content each card can show.
